src/enum_net.cpp

```diff
@@ -552,54 +552,55 @@ namespace libtorrent
 			GetAdaptersAddresses_t GetAdaptersAddresses = (GetAdaptersAddresses_t)GetProcAddress(
 				iphlp, "GetAdaptersAddresses");
 
-			if (GetAdaptersAddresses)
+			if (GetAdaptersAddresses == NULL)
 			{
-				PIP_ADAPTER_ADDRESSES adapter_addresses = 0;
-				ULONG out_buf_size = 0;
-				if (GetAdaptersAddresses(AF_UNSPEC, GAA_FLAG_SKIP_MULTICAST | GAA_FLAG_SKIP_DNS_SERVER
-					| GAA_FLAG_SKIP_ANYCAST, NULL, adapter_addresses, &out_buf_size) != ERROR_BUFFER_OVERFLOW)
-				{
-					FreeLibrary(iphlp);
-					ec = asio::error::operation_not_supported;
-					return std::vector<ip_interface>();
-				}
+				FreeLibrary(iphlp);
+				ec = error_code(boost::system::errc::not_supported, generic_category());
+				return std::vector<ip_interface>();
+			}
 
-				adapter_addresses = (IP_ADAPTER_ADDRESSES*)malloc(out_buf_size);
-				if (!adapter_addresses)
-				{
-					FreeLibrary(iphlp);
-					ec = asio::error::no_memory;
-					return std::vector<ip_interface>();
-				}
+			ULONG buf_size = 10000;
```

Comment:
I guess you already know this, quoting MSDN:
```
The recommended method of calling the GetAdaptersAddresses function is to pre-allocate a 15KB working buffer pointed to by the AdapterAddresses parameter. 
```
Not sure should you really follow that...